feat: [SIW-1967] Align trust chain implementation with 0.9.0

ale-mazz opened the pull request:
<!--- Please always add a PR description as if nobody knows anything about the context these changes come from. -->
<!--- Even if we are all from our internal team, we may not be on the same page. -->
<!--- Write this PR as you were contributing to a public OSS project, where nobody knows you and you have to earn their trust. -->
<!--- This will improve our projects in the long run! Thanks. -->

#### List of Changes

<!--- Describe your changes in detail -->
- Added a new function, `buildTrustChain,` to build a Trust Chain, which will then be verified with the `verifyTrustChain` function.
- Fixed comments
- Aligned Federation types with 0.9.0

> [!NOTE]  
> x.509 certificates validation will be added later inside another PR.

#### Motivation and Context

<!--- Why is this change required? What problem does it solve? -->
This PR implements the changes needed for the alignment with [0.9.0 technical specifications](https://italia.github.io/eudi-wallet-it-docs/v0.9.0/en/trust.html)

#### How Has This Been Tested?

> [!NOTE]  
> `buildTrustChain` cannot be tested until the provider releases a valid Trust Anchor.

<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->

#### Screenshots (if appropriate):

<!--- Attach screenshots in case changes impact UI. -->

#### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

gispada commented on src/trust/index.ts:
```diff
@@ -259,3 +262,133 @@ export async function getSignedEntityStatement(
+    .then((res) => res.text())
+    .then((text) => {
+      try {
+        return JSON.parse(text) as string[];
```
Instead of having a `try/catch` within a `then` block, why not using `Zod` with `safeParse`?

```suggestion
    const federationList = await fetch().then(FederationListResponse.safeParse);
    if (!federationList.success) // throw error
```

ale-mazz replied:
Thanks, @gispada. I am still not used to Zod, and I forgot to add the validation even if it is a simple type.

gispada commented on src/trust/chain.ts:
```diff
@@ -114,38 +85,51 @@ export async function validateTrustChain(
+          entityConfigurationResult.data.payload.iss,
+          { appFetch }
+        );
+      } else if (entityStatementResult.success) {
```
Since we are returning early we can remove the `else if` and just use `if`, what do you think?

grausof reviewed (approved):
It seems ok for now. Given the complexity of the code, it should be tried on a test service. But what do you think of adding unit tests perhaps using non -regulatory examples?

ale-mazz commented:
> It seems ok for now. Given the complexity of the code, it should be tried on a test service. But what do you think of adding unit tests perhaps using non -regulatory examples?

Yes its a great idea, working on them right now. I'll update this PR with those tests!

ale-mazz commented:
@grausof So far I have
- Refactored `buildTrustChain` since it was not acting correctly
- I built some tests and mocked the HTTP response; everything should be aligned with what we expect.
- Fixed `zod` types
- Fixed mocks that weren't aligned with the spec.

Let me know if everything looks okay. Thanks!

grausof commented:
Great 🎉
